### PR TITLE
Create JavaScript File to Host Game Logic

### LINKED ISSUE #20: Create Javascript file to host game logic
## Summary:
We need a Javascript file to host all the logic for the game; logic gate calculations, changes to life bar, game progress, etc.  Should be essentially a blank page.

### PR DESCRIPTION
**Pull request by VictorSohier**
## Changes:

Added JavaScript File(s) to host application logic

## Screenshots:
PR Screen Shot

## Impacts and Notes:
Well, we now have a rough framework to build the rest of the application off of!

## Checklist:
- [x] tested locally
- [x] added new dependencies
- [x] updated the docs
- [ ] added a test

### CODE REVIEW

**Comment by rodneybarnes** [comment]
Let's hold off on merging this for now as I'll be doing a sort of public review of this on Friday (with permission, of course!)

**Review by iferc — COMMENTED** [review]
Just wanted to mention that it can be a good practice use `const` over `let` whenever defining something that won't change (like a function). The most common advantage of it using `const` over `let` is that any attempt to change the value that a `const` variable name references will throw an exception, whereas a `let` variable can be changed.

It is also important to note that this only applies to the variable as a whole and not array items or object properties contained within. [Object freezing](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze) is used to make the contents of those types of values immutable/unchanging as well.

Specifying that some piece of data is immutable is a bit like having a safety net in our code. If ever any of us write some code that unintentionally changes a value that we explicitly never expected would change, we immediately get alerted to that fact and can correct it before it becomes a larger bug down the road.

Some more information and examples at https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/const

Nice boolean logic gate graphics! I can't wait to see these in action!

**Comment by VictorSohier** [comment]
> Just wanted to mention that it can be a good practice use `const` over `let` whenever defining something that won't change (like a function). The most common advantage of it using `const` over `let` is that any attempt to change the value that a `const` variable name references will throw an exception, whereas a `let` variable can be changed.
> 
> It is also important to note that this only applies to the variable as a whole and not array items or object properties contained within. [Object freezing](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze) is used to make the contents of those types of values immutable/unchanging as well.
> 
> Specifying that some piece of data is immutable is a bit like having a safety net in our code. If ever any of us write some code that unintentionally changes a value that we explicitly never expected would change, we immediately get alerted to that fact and can correct it before it becomes a larger bug down the road.
> 
> Some more information and examples at https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/const
> 
> Nice boolean logic gate graphics! I can't wait to see these in action!

Understood, I come from using a readonly modifier in C#. Probably will change that when I can. Appreciate the comments on the sprite work. That will probably go into a spritesheet.

File: docs/index.js.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+## How it functions
```
Comment (tndmohawk) [inline docs/index.js.md:1]: This is awesome 🥳

File: docs/index.js.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+
+P5.js operates on a setup/draw system and each scene is a function meant to act
```
Comment (tndmohawk) [inline docs/index.js.md:3]:
```suggestion
P5.js operates on a setup/draw system and each scene is a function meant to act TEST
```

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+//P5.js documentation: https://p5js.org/
+//P5.js tutorials and examples by Daniel Shiffman: https://www.youtube.com/watch?v=yPWkPOfnGsw&list=PLRqwX-V7Uu6Zy51Q-x9tMWIv9cueOFTFA&index=2
+//And if you haven't heard of Daniel Shiffman, watch his stuff. It is either
+//done in processing or p5.js of which one is a port of the other
+
+let scenes = {
```
Comment (tndmohawk) [inline src/index.js]:
```suggestion
const scenes = {
```

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    "Challenge Mode": challengeMode,
+    "Cheat Sheet": cheatSheet,
+    "Game Over": gameOver,
+    "Instructions": instructions,
+    "Main Menu": mainMenu,
+    "Practice Mode": practiceMode,
+    "Settings": settings
+};
+
+let currentScene = "Main Menu";
```
Comment (tndmohawk) [inline src/index.js:16]:
```suggestion
const currentScene = "Main Menu";
```

File: src/index.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+function setup()
+{
+    createCanvas(640,480);
+}
+
+function draw()
```
Comment (tndmohawk) [inline src/index.js:30]: Should we document what this function does, and how everything ties together?

**Review by tndmohawk — COMMENTED** [review]
Really great work here! As discussed, it would be nice to have some comments around the functions in `index.js` to help the rest of us understand how everything is tied together.

**Review by tndmohawk — APPROVED** [review]
Awesome!!!!

**Pull request merged**